app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php

```diff
@@ -58,13 +59,14 @@ public static function identifyLeadsCompany($parameters, Lead $lead, CompanyMode
             $country = isset($parameters['country']) ? $parameters['country'] : null;
             $state   = isset($parameters['state']) ? $parameters['state'] : null;
 
-            $company = $companyRepo->identifyCompany($companyName, $city, $country, $state);
+            $companyEntity = $companyRepo->identifyCompany($companyName, $city, $country, $state);
```

Comment:
Not quite what I meant. `identifyCompany` returns an array (not an entity) which will now break https://github.com/mautic/mautic/pull/2922/files#diff-9b15b6be04482d6f57b012a98d72e58aR846 since it's expecting an entity. For this change to work, `identifyCompany` needs to return an entity (the whole method needs to be changed to DQL) and then the `$company` array hydrated with the equivalents from the entity's getter methods.